docs/source/algorithms/RefineSatellitePeaks-v1.rst

```diff
@@ -0,0 +1,80 @@
+.. algorithm::
+
+.. summary::
+
+.. properties::
+
+Description
+-----------
+
+RefineSatellitePeaks can be used to refine the locations of "satellite" peaks
+occurring at fractional HKL locations in reciprocal space. RefineSatellitePeaks
+takes a :ref:`MDWorkspace <MDWorkspace>` of experimental data, a
+:ref:`PeaksWorkspace <PeaksWorkspace>` containing the locations of peaks with
+integer HKL, and another PeaksWorkspace containing a subset of peaks found at
+fractional HKL. 
+
+For each satellite peak the euclidean distance between the nearest integer peak
+and satellite are computed in the HKL frame. Peaks are then grouped according
+to euclidean distance using using the properties `NumOfQs` and
+`ClusterThreshold`. If `NumOfQs` is specified then each offset will be grouped
+into exactly `k` clusters. If `ClusterThreshold` is specified then offsets will
+be grouped into clusters seperated by no more than a cophenetic distance below
+this threshold.  The centroid of each cluster calculated for each group and is
+used as the offset to predict the location of fractional peaks everywhere in
+the :ref:`MDWorkspace <MDWorkspace>`.
+
+For each predicted fractional peak, the local centroid (the radius of which is
+defined by `PeakRadius`) in the MD data is found and this is taken as the actual
+position of the satellite peaks.
+
+Finally the found satellite peaks are integerated using the
+:ref:`IntegratePeaksMD <algm-IntegratePeaksMD-v2>` algorithm with the parameters `PeakRadius`,
+`BackgroundInnerRadius`, and `BackgroundOuterRadius`. Satellite peaks are
+discarded if there I/sigma value is less than the parameter `IOverSigma`.
+
```

Comment:
Minor point. In the description, you specify the inputs, but now what you actually output. Is it enough to look at the property outputs?

Reply:
Also, I presume there were discussions about whether `IntegratePeaksMD` was the right solution, over say `IntegerateEllipsoids` - Are the peaks typically spherical in HKL?

Reply:
The integration step is only there to help discard anything that is predicted that is not really a peak. As these peaks are likely to be weak, the integrated value should be taken with a pinch of salt, reintegrating them with more carefully chosen parameters is probably essential. For the purpose of deciding whether a peak exists or not spherical integration is probably enough.

I can't add multiple integration approaches because annoyingly the different integration algorithms take different workspace types as input (`IntegrateEllipsoids` needs a matrix workspace which it converts events to HKL space).

As this is not clear I suggest that the documentation should be changed to highlight this. Another idea would be to manually set all the intensities back to zero after integrating so the user is forced to reintegrate the peaks.